### ISSUE TITLE
Update to FakeItEasy 3.1.0 breaks tests

### ISSUE THREAD

**Issue by benArrayx** [jordimontana82/fake-xrm-easy#143]
Hi Jordi, I was updating all my NuGet packages today which showed updates available, FakeItEasy was updated to 3.1.0 but then on running tests I got this error: 

Result Message:	System.MissingMethodException : Method not found: 'FakeItEasy.Configuration.IAfterCallSpecifiedWithOutAndRefParametersConfiguration FakeItEasy.ReturnValueConfigurationExtensions.ReturnsLazily(FakeItEasy.Configuration.IReturnValueConfiguration`1<!!0>, System.Func`4<!!1,!!2,!!3,!!0>)'.

Downgrading to FakeItEasy 2.3.3 resolved this issue. I'm not sure if you can do anything about this by specifying the latest compatible version in your NuGet package?

Regards, Ben

**Comment by jordimontana82** [comment]
Hi Ben,

Thanks for this... yeah, in theory the last one I used is referenced in the package dependencies.

![image](https://cloud.githubusercontent.com/assets/3968950/24305870/66316576-10b6-11e7-86bf-7a1d356e8e1a.png)

But those are just greater than or equal dependencies... i.e. , you can't know if upgrading to a newer version will break anything and I'm not sure we can't do anything different than just recommending to use the same versions as defined in the package dependency by default.

**Comment by benArrayx** [comment]
I’ve seen other packages that define a specific version, for example LibGit2Sharp https://www.nuget.org/packages/LibGit2Sharp which has a dependency to LibGit2Sharp.NativeBinaries (= 1.0.164) – this doesn’t let you update LibGit2Sharp.NativeBinaries past a compatible version.

Regards, Ben

**Comment by jordimontana82** [comment]
Nice one.

We could probably enforce that specific version then :+1:

**Comment by benArrayx** [comment]
Hi Jordi, I have been experiencing some crazy problems when running tests in bulk, primarily when running an ExecuteMultipleRequest and trying to cast the return to an ExecuteMultipleResponse, getting the error 'unable to cast Castle.Proxies.OrganisationResponseProxy to ExecuteMultipleResponse' (not exact wording), and also on doing a simple Retrieve being unable to case an object of type Castle.Proxies.EntityProxy to Entity.

These errors do not *ever* manifest themselves when running tests individually. They *always* manifest when running certain combinations of tests, for example I have 2 tests which when run together will always have at least one failures, but the failures are not reliable, sometimes 1 test fails, sometimes the other, sometimes both. But, they always fail on not being able to cast a Castle.Proxy object.

I finally updated to FakeItEasy 2.3.3 (I had reverted earlier to 1.25.3) in an attempt to resolve this after spending several hours trying to debug the issue and failing to get anywhere. Thankfully, it has fixed the issue! As such I definitely recommend you to do this update.
